server/schedule/placement/rule_manager.go

```diff
@@ -437,5 +442,148 @@ func (m *RuleManager) DeleteRuleGroup(id string) error {
 	defer m.Unlock()
 	p := m.ruleConfig.beginPatch()
 	p.deleteGroup(id)
-	return m.tryCommitPatch(p)
+	if err := m.tryCommitPatch(p); err != nil {
+		return err
+	}
+	log.Info("group config reset", zap.String("group", id))
+	return nil
+}
+
+// GetAllGroupBundles returns all rules and groups configuration. Rules are
+// grouped by groups.
+func (m *RuleManager) GetAllGroupBundles() []GroupBundle {
+	m.RLock()
+	defer m.RUnlock()
+	var bundles []GroupBundle
+	for _, g := range m.ruleConfig.groups {
+		bundles = append(bundles, GroupBundle{
+			ID:       g.ID,
+			Index:    g.Index,
+			Override: g.Override,
+		})
+	}
+	for _, r := range m.ruleConfig.rules {
+		for i := range bundles {
+			if bundles[i].ID == r.GroupID {
+				bundles[i].Rules = append(bundles[i].Rules, r)
+			}
+		}
+	}
+	sort.Slice(bundles, func(i, j int) bool {
+		return bundles[i].Index < bundles[j].Index ||
+			(bundles[i].Index == bundles[j].Index && bundles[i].ID < bundles[j].ID)
+	})
+	for _, b := range bundles {
+		sortRules(b.Rules)
+	}
+	return bundles
+}
+
+// GetGroupBundle returns a group and all rules belong to it.
+func (m *RuleManager) GetGroupBundle(id string) (b GroupBundle) {
+	m.RLock()
+	defer m.RUnlock()
+	b.ID = id
+	if g := m.ruleConfig.groups[id]; g != nil {
+		b.Index, b.Override = g.Index, g.Override
+	}
+	for _, r := range m.ruleConfig.rules {
```

Comment:
Is there a need to filter rules, if the specified group is empty/non-exist? A similar problem raises on `SetGroupBundle`, and `DeleteGroupBundle`: If it is not a regexp request, a check can be dome by a simple map operation.

Reply:
Good point.